lwr_controllers/include/lwr_controllers/gravity_compensation.h

```diff
@@ -0,0 +1,41 @@
+#ifndef GRAVITY_COMPENSATION_H
+#define GRAVITY_COMPENSATION_H
+
+#include "KinematicChainControllerBase.h"
+
+#include <std_msgs/Bool.h>
+#include <std_msgs/Float32.h>
+
+namespace lwr_controllers
+{
+    class GravityCompensation: public controller_interface::KinematicChainControllerBase<hardware_interface::PositionJointInterface>
```

Comment:
A good reason to parse the URDF here is if you actually send the f_dyn computed from the URDF mode, including the tool, which actually would be a proper gravity compensation controller.